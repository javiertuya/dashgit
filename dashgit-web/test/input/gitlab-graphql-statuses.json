{
    "data": {
        "projects": {
            "nodes": [
                {
                    "id": "gid://gitlab/Project/5", "fullPath": "org1/proj5", "lastActivityAt": "2024-02-05T15:49:00+01:00",
                    "webUrl": "https://mygitlab.com/org1/proj5",
                    "pipelines": {
                        "nodes": [
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6550", "startedAt": "2024-02-05T15:50:00+01:00", "finishedAt": null,
                                "latest": true, "refPath": "refs/heads/building", "name": null, "status": "RUNNING"
                            },
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6549", "startedAt": "2024-02-05T15:39:00+01:00", "finishedAt": "2024-02-05T15:49:00+01:00",
                                "latest": true, "refPath": "refs/heads/feature", "name": null, "status": "SUCCESS"
                            },
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6548", "startedAt": "2024-02-10T15:38:00+01:00", "finishedAt": "2024-02-05T15:48:00+01:00",
                                "latest": true, "refPath": "refs/heads/main", "name": null, "status": "FAILED"
                            },
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6540", "startedAt": "2024-02-05T15:30:00+01:00", "finishedAt": "2024-02-05T15:40:00+01:00",
                                "latest": true, "refPath": "refs/heads/feature", "name": null, "status": "FAILED"
                            }
                        ]
                    },
                    "mergeRequests": {
                        "nodes": [
                            {
                                "iid": "50", "webUrl": "https://mygitlab.com/org1/proj5/-/merge_requests/50",
                                "sourceBranch": "building", "title": "pr branch building now"
                            },
                            {
                                "iid": "49", "webUrl": "https://mygitlab.com/org1/proj5/-/merge_requests/49",
                                "sourceBranch": "feature", "title": "pr branch with pipeline"
                            }
                        ]
                    }
                },
                {
                    "id": "gid://gitlab/Project/7", "fullPath": "org2/proj7", "lastActivityAt": "2024-02-07T17:49:00+01:00",
                    "webUrl": "https://mygitlab.com/org2/proj7",
                    "pipelines": {
                        "nodes": [
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6749", "startedAt": "2024-02-07T17:39:00+01:00", "finishedAt": "2024-02-07T17:49:00+01:00",
                                "latest": true, "refPath": "refs/heads/hidden1", "name": null, "status": "SUCCESS"
                            },
                            { 
                                "id": "gid://gitlab/Ci::Pipeline/6748", "startedAt": "2024-02-07T17:38:00+01:00", "finishedAt": "2024-02-07T17:48:00+01:00",
                                "latest": true, "refPath": "refs/heads/hidden2", "name": null, "status": "SUCCESS"
                            }
                        ]
                    },
                    "mergeRequests": {
                        "nodes": [
                            {
                                "iid": "49", "webUrl": "https://mygitlab.com/org2/proj7/-/merge_requests/49",
                                "sourceBranch": "branch1", "title": "pr branch without pipeline"
                            },
                            {
                                "iid": "48", "webUrl": "https://mygitlab.com/org2/proj7/-/merge_requests/48",
                                "sourceBranch": "hidden1", "title": "hidden pr branch with pipeline"
                            },
                            {
                                "iid": "47", "webUrl": "https://mygitlab.com/org2/proj7/-/merge_requests/47",
                                "sourceBranch": "hidden3", "title": "hidden pr branch without pipeline"
                            }
                        ]
                    }
                }
            ]
        }
    }
}