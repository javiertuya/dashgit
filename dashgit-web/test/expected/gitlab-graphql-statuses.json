{
  "header": {
    "provider": "GitLab",
    "uid": "0-gitlab",
    "user": "usr1",
    "url": "https://mygitlab.com",
    "repo_names": [
      "org1/proj5",
      "org2/proj6",
      "org2/proj7"
    ],
    "target": "",
    "message": ""
  },
  "items": [
    {
      "repo_name": "org1/proj5",
      "type": "pr",
      "iid": "49",
      "branch_name": "feature",
      "status": "success",
      "title": "pr branch with pipeline",
      "actions": {},
      "author": "",
      "assignees": "",
      "created_at": "2024-02-05T15:39:00+01:00",
      "updated_at": "2024-02-05T15:49:00+01:00",
      "iidstr": "",
      "url": "https://mygitlab.com/org1/proj5/-/merge_requests/49",
      "branch_url": "https://mygitlab.com/org1/proj5/-/tree/feature",
      "repo_url": "https://mygitlab.com/org1/proj5",
      "labels": [],
      "gid": "gid://gitlab/Project/5",
      "uid": "org1-proj5_pr_49"
    },
    {
      "repo_name": "org1/proj5",
      "type": "branch",
      "iid": "",
      "branch_name": "main",
      "status": "failure",
      "title": "",
      "actions": {},
      "author": "",
      "assignees": "",
      "created_at": "2024-02-10T15:38:00+01:00",
      "updated_at": "2024-02-05T15:48:00+01:00",
      "iidstr": "",
      "url": "",
      "branch_url": "https://mygitlab.com/org1/proj5/-/tree/main",
      "repo_url": "https://mygitlab.com/org1/proj5",
      "labels": [],
      "gid": "gid://gitlab/Project/5",
      "uid": "org1-proj5_branch_main"
    },
    {
      "repo_name": "org1/proj5",
      "type": "pr",
      "iid": "50",
      "branch_name": "building",
      "status": "pending",
      "title": "pr branch building now",
      "actions": {},
      "author": "",
      "assignees": "",
      "created_at": "2024-02-05T15:50:00+01:00",
      "updated_at": "2024-02-05T15:50:00+01:00",
      "iidstr": "",
      "url": "https://mygitlab.com/org1/proj5/-/merge_requests/50",
      "branch_url": "https://mygitlab.com/org1/proj5/-/tree/building",
      "repo_url": "https://mygitlab.com/org1/proj5",
      "labels": [],
      "gid": "gid://gitlab/Project/5",
      "uid": "org1-proj5_pr_50"
    },
    {
      "repo_name": "org2/proj6",
      "type": "branch",
      "iid": "",
      "branch_name": "main",
      "status": "notavailable",
      "title": "",
      "actions": {},
      "author": "",
      "assignees": "",
      "created_at": "",
      "updated_at": "",
      "iidstr": "",
      "url": "",
      "branch_url": "",
      "repo_url": "https://mygitlab.com/org2/proj6",
      "labels": [],
      "gid": "gid://gitlab/Project/6",
      "uid": "org2-proj6_branch_main"
    },
    {
      "repo_name": "org2/proj7",
      "type": "pr",
      "iid": "49",
      "branch_name": "branch1",
      "status": "notavailable",
      "title": "pr branch without pipeline",
      "actions": {},
      "author": "",
      "assignees": "",
      "created_at": "",
      "updated_at": "",
      "iidstr": "",
      "url": "https://mygitlab.com/org2/proj7/-/merge_requests/49",
      "branch_url": "",
      "repo_url": "https://mygitlab.com/org2/proj7",
      "labels": [],
      "gid": "gid://gitlab/Project/7",
      "uid": "org2-proj7_pr_49"
    }
  ]
}